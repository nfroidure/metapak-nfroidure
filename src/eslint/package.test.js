'use strict';

const packageTransformer = require('./package');

describe('eslint', () => {
  describe('Package transformer', () => {
    it('should work with some files', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              files: 'src/*.js src/**/*.js',
            },
          },
        })
      ).toMatchSnapshot();
    });

    it('should work with child packages', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              childPackage: true,
              files: 'src/*.js src/**/*.js',
            },
          },
        })
      ).toMatchSnapshot();
    });

    it('should work with typescript configs', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              childPackage: true,
              files: 'src/*.js src/**/*.js',
            },
            configs: ['typescript'],
          },
        })
      ).toMatchSnapshot();
    });
  });
});
