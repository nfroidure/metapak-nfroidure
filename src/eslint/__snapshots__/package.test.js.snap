// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint Package transformer should work with some files 1`] = `
Object {
  "devDependencies": Object {
    "eslint": "^5.6.0",
    "eslint-plugin-prettier": "^2.6.2",
    "prettier": "^1.14.2",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "eslint",
      "eslint-config-prettier",
      "prettier",
    ],
  },
  "metapak": Object {
    "data": Object {
      "files": "src/*.js src/**/*.js",
    },
  },
  "scripts": Object {
    "lint": "eslint src/*.js src/**/*.js",
    "prettier": "prettier --write src/*.js src/**/*.js",
  },
}
`;
